executorCwl/src/main/scala/dx/executor/cwl/CwlWorkflowExecutor.scala

```diff
@@ -372,7 +372,7 @@ case class CwlWorkflowExecutor(workflow: Workflow,
             if (stepInput.linkMerge.nonEmpty || stepInput.pickValue.nonEmpty) {
               val mergedValues = (sourceValues, stepInput.linkMerge) match {
                 case (_, Some(LinkMergeMethod.MergeFlattened)) =>
-                  sourceValues.flatMap {
+                  sourceValues.map(s => (CwlOptional.unwrapOptional(s._1), s._2)).flatMap {
```

Comment:
`sourceValues` could be an array of optional items. So without `unwrapOptional` those sources with type `CwlOptional(t)` will not match the `case (array: CwlArray)` below.